### ISSUE TITLE
The question about the mask

### ISSUE THREAD

**Issue by quelladai** [SeitaroShinagawa/chainer-partial_convolution_image_inpainting#13]
I am sorry to trouble you. I want to know what is the meaning of mask_b (https://github.com/SeitaroShinagawa/chainer-partial_convolution_image_inpainting/blob/master/common/net.py#L62) in your code  and where did you pixel-wise multiplicate an input image and mask to simulate a broken image? 
And the size of the mask is as the same as the feature values x, which is also the same as the cobvolution sliding windows, why did you say that set the mask window as 1(https://github.com/SeitaroShinagawa/chainer-partial_convolution_image_inpainting/blob/master/common/net.py#L55)?
Besides, I can not understand the unmasking rule?
I am sorry that i have so many questions, hope your responce!

**Comment by quelladai** [comment]
And what is the meaning of maskW(https://github.com/SeitaroShinagawa/chainer-partial_convolution_image_inpainting/blob/master/common/net.py#L46)

**Comment by SeitaroShinagawa** [comment]
Hi @quelladai,  
Thank you for your interest.   

mask_b is the mask value in boolean type. It consists of boolean 0 or 1. It works as the condition of F.where in L.64 and L.68. F.where represents "chainer.functions.where" which works same as "numpy.where" function. 
https://docs.chainer.org/en/stable/reference/generated/chainer.functions.where.html

Pixel-wise multiplication between an input image and mask is here,
https://github.com/SeitaroShinagawa/chainer-partial_convolution_image_inpainting/blob/master/common/net.py#L56

The reason why I set the mask window as 1 is to calculate (1) and (2) in the original paper.
https://arxiv.org/abs/1804.07723
In "3.2 Network Architecture and Implementation--Implementation," the authors says,
"The straightforward implementation is to define binary masks of size C×H×W, the same size with their associated images/features, and then to implement mask updating is implemented using a **fixed convolution layer**, with the same kernel size as the partial convolution operation, but with **weights identically set to 1 and bias set to 0**."

Namely, maskW represents the fixed weight to calculate sum(M).

**Comment by quelladai** [comment]
Thank you very much

**Comment by micklexqg** [comment]
@SeitaroShinagawa ,
your mask update is as follow:
new_mask = torch.ones_like(output)
        new_mask = new_mask.masked_fill_(no_update_holes, 0.0)
it means you update the mask with 0 while the original paper update it with 1.
 Is the mask in your code is inverse to the original paper?
